src/shared/codelens/localLambdaRunner.ts

```diff
@@ -357,64 +363,92 @@ export const makeBuildDir = async (): Promise<string> => {
 
 export async function makeInputTemplate(params: {
     baseBuildDir: string,
-    codeDir: string,
+    templateDir: string,
     documentUri: vscode.Uri
     originalHandlerName: string,
     handlerName: string,
     runtime: string,
     workspaceUri: vscode.Uri,
-}): Promise<string> {
+    channelLogger: ChannelLogger
+}): Promise<InputTemplateInfo> {
+
+    // Switch over to the output channel so the user has feedback that we're getting things ready
+    params.channelLogger.channel.show(true)
+
+    params.channelLogger.info(
+        'AWS.output.sam.local.start',
+        'Preparing to run {0} locally...',
+        params.handlerName
+    )
+
     const inputTemplatePath: string = path.join(params.baseBuildDir, 'input', 'input-template.yaml')
     ExtensionDisposableFiles.getInstance().addFolder(inputTemplatePath)
 
     // Make function handler relative to baseDir
     const handlerFileRelativePath = path.relative(
-        params.codeDir,
+        params.templateDir,
         path.dirname(params.documentUri.fsPath)
     )
 
     const workspaceFolder = vscode.workspace.getWorkspaceFolder(params.workspaceUri)
+    let existingLambda: LocalLambda | undefined
     let existingTemplateResource: CloudFormation.Resource | undefined
     if (workspaceFolder) {
 
-        const relativeOriginalFunctionHandler = normalizeSeparator(
-            path.join(
-                handlerFileRelativePath,
-                params.originalHandlerName,
+        let relativeOriginalFunctionHandler = params.originalHandlerName
+        if (getFamily(params.runtime) !== SamLambdaRuntimeFamily.DotNet) {
```

Comment:
We discussed in person that it looks like there is logic that should be the domain of the caller in here.

Looking at this method, there are some wins to be had by simplifying it:
* Callers can determine their function handler before calling this method (addresses "interpreted runtimes = relative path + lambda handler" vs "dotnet = predefined handler string xxx:xxx.yyy:zzz")
* Callers can determine where the corresponding template file is, and what the matching template resource is. The matching template resource can be passed into this method.
* Callers can determine the codeuri location and pass that directly in. (allows code providers to handle additional edge cases as necessary)

Some of the wins:
* reduced function param list
* increased testability
* broken into more reusable components

Reply:
✔️